### PR TITLE
fix error during visualization on explanation with id for eval_y_predicted and eval_y_predicted_proba in datasets mixin

### PR DESCRIPTION
**Pull request by imatiach-msft**
fix error during visualization on explanation with id for eval_y_predicted and eval_y_predicted_proba in datasets mixin

### CODE REVIEW

File: python/interpret_community/explanation/explanation.py [hunk 1]
```diff
@@ -513,9 +513,12 @@ def _local_data(self, parent_data, key=None):
         # Note: we currently don't have access to instances on explanation
         parent_data[InterpretData.VALUES] = None
         if key is not None:
+            # Note: eval_y_predicted and eval_data can be id, in which case we should skip this logic
+            has_eval_y_predicted = not isinstance(self._eval_y_predicted, str)
+            has_eval_data = not isinstance(self._eval_data, str)
             # Note: the first argument should be the true y's but we don't have that
             # available currently, using predicted instead for now
-            if _DatasetsMixin._does_quack(self):
+            if _DatasetsMixin._does_quack(self) and has_eval_y_predicted and has_eval_data:
```
Comment (gregorybchris) [inline python/interpret_community/explanation/explanation.py:521]: Should this check be part of `_does_quack`? Does the check for str type apply to all `_DatasetsMixin`?

File: python/interpret_community/explanation/explanation.py [hunk 2]
```diff
@@ -1215,15 +1225,15 @@ def _convert_to_list(self, data):
 
         :param data: The data to be converted.
         :type data: np.array, pd.DataFrame, list, scipy.sparse
-        :return: The data converted to a list (except for sparse which is unchanged).
-        :rtype: list | scipy.sparse | list[scipy.sparse]
+        :return: The data converted to a list (except for sparse or string which is unchanged).
```
Comment (gregorybchris) [inline python/interpret_community/explanation/explanation.py:1228]: Might want to clarify that string in this case means Dataset reference?

File: python/interpret_community/explanation/explanation.py (hunk 1 above)
Comment (imatiach-msft) [reply python/interpret_community/explanation/explanation.py:521]: does quack just checks those properties exist, not what the values are.  This change is in the data method.

File: python/interpret_community/explanation/explanation.py (hunk 2 above)
Comment (imatiach-msft) [reply python/interpret_community/explanation/explanation.py:1228]: done!  changed to dataset ID

File: python/interpret_community/explanation/explanation.py (hunk 1 above)
Comment (gregorybchris) [reply python/interpret_community/explanation/explanation.py:521]: I think it might be good to isolate the logic checking the type of these two properties. These two properties are always related to a `_DatasetsMixin` right? Maybe `_DatasetsMixin` can have another static method that checks whether the dataset is a reference or in-memory data? I imagine something like `_DatasetsMixin.is_reference()`

Comment (gregorybchris) [reply python/interpret_community/explanation/explanation.py:521]: Then in plain English you would say "only update parent_data if this adheres to the `_DatasetsMixin` interface AND it's not a reference type `_DatasetsMixin`"

Comment (imatiach-msft) [reply python/interpret_community/explanation/explanation.py]: done! added _is_reference() method

**Review by gregorybchris — APPROVED** [review]
Thanks for making that change!
